python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -50,15 +50,6 @@ def _create_dd_meta(cls, dataset_info, **kwargs):
             kwargs.get("schema", None),
         )
 
-        # If `strings_to_categorical==True`, convert objects to int32
-        strings_to_cats = kwargs.get("strings_to_categorical", False)
-        for col in meta_cudf._data.names:
-            if (
-                isinstance(meta_cudf._data[col], cudf.core.column.StringColumn)
-                and strings_to_cats
-            ):
-                meta_cudf._data[col] = meta_cudf._data[col].astype("int32")
```

Comment:
I know that some RecSys-based users have historically relied on `strings_to_categorical=True` to deal with the fact that their ML/DL model would often require them to convert string data to numerical data (but they needed to preserve information in the original string data on disk, and couldn't work with a fully-numerical Parquet file).

What is the new "best practice" for these users? Should we suggest something like `df["A"] = df["A"].hash_values()`? Was `strings_to_categorical=True` previously doing something clever to avoid reading the entire string column into GPU memory?

Reply:
cc @EvenOldridge (in case you had further input here)

Reply:
> Was `strings_to_categorical=True` previously doing something clever to avoid reading the entire string column into GPU memory?

@vuule would be the right one to know if there was any such thing happening.



> What is the new "best practice" for these users? Should we suggest something like `df["A"] = df["A"].hash_values()`?

If someone was relying on this, yes `hash_values` would be the consistent alternative. `hash_values` by itself is consistent, but won't yield similar integers as to legacy `strings_to_categorical=True` behavior.

Reply:
> Was strings_to_categorical=True previously doing something clever to avoid reading the entire string column into GPU memory?

We don't create a string column with this option - string data in the file is hashed in the kernel and we just return an int32 column. So, yeah, something like df["A"] = df["A"].hash_values() is likely to have higher memory use at times.

Reply:
>We don't create a string column with this option - string data in the file is hashed in the kernel and we just return an int32 column. So, yeah, something like df["A"] = df["A"].hash_values() is likely to have higher memory use at times.

That's too bad, but certainly makes sense. I don't think anyone was actually **using** `strings_to_categorical`, but I get the sense that people will be starting to ask for exactly this functionality in the near future :/

Reply:
If there are practical use cases, I wouldn't mind the feature as a "read strings as int" conversion support.

Reply:
Thanks @vuule, that's good to know. I'll raise an issue if/when I have a practical use case to point to (otherwise this is low priority).